backend/app/models/event.rb

```diff
@@ -5,7 +5,7 @@ class Event < ActiveRecord::Base
   belongs_to :enrollment
 
   belongs_to :user, optional: true
-  validates :user, presence: true, if: proc { |event| event.name != "reminder" }
+  validates :user, presence: true, if: proc { |event| event.name != "reminder" && event.name != "archive" }
```

Comment:
```suggestion
  validates :user, presence: true, if: proc { |event| %w[reminder archive].exclude?(event.name) }
```

You should make this change in a single commit imo